tools/lsp/preview/element_selection.rs

```diff
@@ -422,3 +455,346 @@ pub fn reselect_element() {
 
     super::set_selected_element(Some(selected), &positions, false);
 }
+
+#[cfg(test)]
+mod tests {
+    use std::path::PathBuf;
+
+    use slint_interpreter::ComponentInstance;
+
+    fn demo_app() -> ComponentInstance {
+        crate::preview::test::compile_test(
+            "fluent",
+            r#"import { Button } from "std-widgets.slint";
+
+component SomeComponent { // 69
+    @children
+}
+
+component Main { // 109
+    width: 200px;
+    height: 200px;
+
+    HorizontalLayout { // 160
+        Rectangle { // 194
+            SomeComponent { // 225
+                Button { // 264
+                    text: "Press me";
+                }
+            }
+        }
+    }
+}
+
+export component Entry inherits Main { /* @lsp:ignore-node */ } // 401
+"#,
+        )
+    }
+
+    #[test]
+    fn test_find_covering_elements() {
+        let component_instance = demo_app();
+
+        let mut covers_center =
+            super::collect_all_element_nodes_covering(100.0, 100.0, &component_instance);
+
+        // Remove the "button" implenmentation details. They must be at the start:
+        let button_path = PathBuf::from("builtin:/fluent-base/button.slint");
+        let first_non_button = covers_center
+            .iter()
+            .position(|sc| {
+                sc.as_element_node().map(|en| en.path_and_offset().0).as_ref() != Some(&button_path)
+            })
+            .unwrap();
+        covers_center.drain(0..first_non_button);
+
+        let test_file = PathBuf::from("/test_data.slint");
+
+        let expected_offsets = [264_u32, 69, 225, 194, 160, 109, 401];
```

Comment:
I find that using hardcoded offsets in the tests is sometimes difficult and not flexible of we want to change the test.
One thing I like to do is to do use `find(` on the source with a comment or element name, to get the byte offset.